### PR TITLE
Add diagrams to document BookCommand

### PR DESCRIPTION
**Pull request by jerremyng**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -233,6 +233,31 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
+### Book Command
```
Comment (cyqjoseph) [inline docs/DeveloperGuide.md:236]: Looks good!

File: docs/diagrams/BookCommandSequenceDiagram.puml
Comment (cyqjoseph) [inline docs/diagrams/BookCommandSequenceDiagram.puml:1]: Great work adding the diagram!

**Review by cyqjoseph — APPROVED** [review]
All good!

File: docs/diagrams/BookCommandSequenceDiagram.puml
Comment (ChrysanthemumT) [reply docs/diagrams/BookCommandSequenceDiagram.puml:1]: yup seems good, last diagram doesn't seem that important of a change, but overall great work!

**Review by ChrysanthemumT — APPROVED** [review]
seems good!

**Pull request merged**